### PR TITLE
[Graphbolt]Adapt negative sampler

### PR DESCRIPTION
**Pull request by peizhou001**
## Description
As we add a new structure to represent unified data structure in graphbolt, change negative sampler to adapt to that.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/negative_sampler.py [hunk 1]
```diff
@@ -41,7 +42,7 @@ def _sample(self, node_pairs):
         """
         Generate a mix of positive and negative samples.
 
-        Parameters
+        node_pair
```
Comment (frozenbugs) [inline python/dgl/graphbolt/negative_sampler.py:45]: revert

File: python/dgl/graphbolt/negative_sampler.py [hunk 2]
```diff
@@ -41,7 +42,7 @@ def _sample(self, node_pairs):
...
         ----------
         node_pairs : Tuple[Tensor] or Dict[etype, Tuple[Tensor]]
```
Comment (frozenbugs) [inline python/dgl/graphbolt/negative_sampler.py:47]: Should the input of the _sample also be LinkPredictionBlock?

Comment (peizhou001) [reply python/dgl/graphbolt/negative_sampler.py:47]: Yes, this will be changed after mini-batch sampler change is completed.

**Review by frozenbugs — APPROVED** [review]

**Comment by frozenbugs** [comment]
@dgl-bot

**Pull request merged**